test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs

```diff
@@ -126,6 +126,76 @@ void CheckValuesSame(bool bl, string tx, float v0, float v1, float v2)
             Assert.Equal(new VectorType(NumberType.R4, 3), schema.GetColumnType(labelIdx));
         }
 
+        private sealed class Obnoxious1
+        {
+            public Scalar<string> Foo { get; }
+            public Vector<float> Bar { get; }
+
+            public Obnoxious1(Scalar<string> f1, Vector<float> f2)
+            {
+                Foo = f1;
+                Bar = f2;
+            }
+        }
+
+        private sealed class Obnoxious2
+        {
+            public Scalar<string> Biz { get; set; }
+            public Vector<double> Blam { get; set; }
+        }
+
+        private sealed class Obnoxious3<T>
+        {
+            public (Scalar<bool> hi, Obnoxious1 my, T friend) Stuff { get; set; }
+        }
+
+        private static Obnoxious3<T> MakeObnoxious3<T>(Scalar<bool> hi, Obnoxious1 my, T friend)
+            => new Obnoxious3<T>() { Stuff = (hi, my, friend) };
+
+        [Fact]
+        public void SimpleTextLoaderObnoxiousTypeTest()
+        {
+            var env = new ConsoleEnvironment(0, verbose: true);
+
+            const string data = "0 hello 3.14159 -0 2\n"
+                + "1 1 2 4 15";
+            var dataSource = new BytesStreamSource(data);
+
+            // Ahhh. No one would ever, ever do this, of course, but just having fun with it.
+
+            void Helper(ISchema thisSchema, string name, ColumnType expected)
+            {
+                Assert.True(thisSchema.TryGetColumnIndex(name, out int thisCol), $"Could not find column '{name}'");
+                Assert.Equal(expected, thisSchema.GetColumnType(thisCol));
+            }
+
+            var text = TextLoader.CreateReader(env, ctx => (
+                yo: new Obnoxious1(ctx.LoadText(0), ctx.LoadFloat(1, 5)),
+                dawg: new Obnoxious2() { Biz = ctx.LoadText(2), Blam = ctx.LoadDouble(1, 2) },
+                how: MakeObnoxious3(ctx.LoadBool(2), new Obnoxious1(ctx.LoadText(0), ctx.LoadFloat(1, 4)),
+                    new Obnoxious2() { Biz = ctx.LoadText(5), Blam = ctx.LoadDouble(1, 10) })));
+
+            var schema = text.AsDynamic.GetOutputSchema();
+            Helper(schema, "yo.Foo", TextType.Instance);
+            Helper(schema, "yo.Bar", new VectorType(NumberType.Float, 5));
+            Helper(schema, "dawg.Biz", TextType.Instance);
+            Helper(schema, "dawg.Blam", new VectorType(NumberType.R8, 2));
+
+            Helper(schema, "how.Stuff.hi", BoolType.Instance);
+            Helper(schema, "how.Stuff.my.Foo", TextType.Instance);
+            Helper(schema, "how.Stuff.my.Bar", new VectorType(NumberType.Float, 4));
```

Comment:
>how.Stuff.my.Bar [](start = 28, length = 16)

'how stuff my bar', hmm...

In the spirit of open-source, should we have a bit more, um, mature naming scheme?..

Reply:
I'll change `Stuff` to `Donut`?

---
In reply to: [222036765](https://github.com/dotnet/machinelearning/pull/1115#discussion_r222036765) [](ancestors = 222036765)